datafusion/expr/src/expr_rewriter/mod.rs

```diff
@@ -336,12 +338,14 @@ impl NamePreserver {
 }
 
 impl SavedName {
-    /// Ensures the name of the rewritten expression is preserved
+    /// Ensures the qualified name of the rewritten expression is preserved
     pub fn restore(self, expr: Expr) -> Result<Expr> {
```

Comment:
The return value no longer needs `Result`, but removing it breaks many optimization rules. Maybe we can handle it in the next PR.